iowrappers/poi_searcher.go

```diff
@@ -144,17 +144,21 @@ func (s *PoiSearcher) NearbySearch(context context.Context, request *PlaceSearch
 	Logger.Debugf("[request_id: %s] number of results from redis is %d", context.Value(ContextRequestIdKey), len(cachedPlaces))
 
 	// update last search time for the city
-	lastSearchTime, cacheMiss := s.redisClient.GetMapsLastSearchTime(context, location, request.PlaceCat)
+	lastSearchTime, lastSearchTimeMiss := s.redisClient.GetMapsLastSearchTime(context, location, request.PlaceCat, request.PriceLevel)
 
 	currentTime := time.Now()
-	// use place data from database if the location is known and the data is fresh, and we have sufficient data
-	if cacheMiss == nil && (currentTime.Sub(lastSearchTime) <= MinMapsResultRefreshDuration && uint(len(cachedPlaces)) >= request.MinNumResults) {
-		Logger.Infof("[request_id: %s] Using Redis to fulfill request. Place Type: %s", context.Value(ContextRequestIdKey), request.PlaceCat)
+	// use place data from the database if the data is fresh, and we have at least one known place satisfying the request
+	if currentTime.Sub(lastSearchTime) <= MinMapsResultRefreshDuration && lastSearchTimeMiss == nil && len(cachedPlaces) > 0 {
```

Comment:
Should we first check (lastSearchTimeMiss == nil) here, in case lastSearchTime returns some invalid result?

Reply:
nit: we can also turn IF condition into a utility function? say `IsCacheResultFresh()`. Then we can make this line shorter and even remove the long comment.

Reply:
Good point of making the IF condition as a utility function. Will do. The reason I am moving the `lastSearchTimeMiss == nil` to the second place is freshness is the top priority regardless of cache miss or not.